pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -168,7 +168,7 @@ func (r *repository) Blobs(ctx context.Context) distribution.BlobStore {
 
 	bs := r.Repository.Blobs(ctx)
 
-	if quotaEnforcing != nil {
+	if !quotaEnforcing.enforcementDisabled {
```

Comment:
:).  After Alexey lands his changes to distribution I have a refactor to the middleware initialization path that prevents us from reloading the clients multiple times (which was not intentional) that will affect some of this code.  But the quick fix here is good.